### PR TITLE
introduce WithErrorHandler/ErrorHandler

### PR DESCRIPTION
**Pull request by amery**
attach error handler to a context.Context

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	if eh != nil {
+		ctx = context.WithValue(ctx, errHandlerKey, eh)
+	}
+
+	return ctx
+}
+
+// ErrorHandler attempts to pull an ErrorHandler from the context.Context
+func ErrorHandler(ctx context.Context) (ErrorHandlerFunc, bool) {
```
[the remark this answers is not in the document]
Comment (amery) [reply error.go:49]: it does check. `eh` will be nil when `!ok`

**Comment by amery** [comment]
@karasz rebased

**Review by karasz — APPROVED** [review]
+1

**Comment by amery** [comment]
`v0.0.2` tagged

**Pull request merged**